File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.HashSet;

class Solution {
    public int solution(int[] nums) {
        int max = nums.length / 2;
        
        HashSet<Integer> set = new HashSet<>();
        
        for(int num : nums) {
            set.add(num);
        }
        
        if(max >= set.size()) return set.size();
        else return max;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    var map = new Map();
    
    for(let num of nums) {
        map.set(num, map.get(num) + 1);
    }
    
    if(map.size > nums.length / 2) return nums.length /2
    else return map.size;
    
}